doskvol-tools/src/components/generator/scoresGenerator/ScoresGenerator.vue

```diff
@@ -0,0 +1,169 @@
+<script setup lang='ts'>
+import {useI18n} from 'vue-i18n'
+import {useScoreStore} from "@/stores/ScoreStore";
+import {storeToRefs} from "pinia";
+import {useScoreGenerator} from "@/components/generator/scoresGenerator/composables/ScoreGeneratorContent";
+import useRandom from "@/composables/Random";
+import type {TableVariant, Variant} from "@/types/TableVariant";
+import BaseGenerator from "@/components/generator/BaseGenerator.vue";
+import BaseTable from "@/components/common/baseTable/BaseTable.vue";
+import GeneratorResultElement from "@/components/generator/GeneratorResultElement.vue";
+
+const {t} = useI18n({useScope: "global"});
+
+const store = useScoreStore();
+const { score } = storeToRefs(store);
+
+const scoreData = useScoreGenerator();
+const { getRandomFromTable, getRandomTableVariant } = useRandom();
+
+const generateRandom = () => {
+  score.value.connectedToFactions = getRandomFromTable(scoreData.getConnectedToFactions());
+  score.value.connectedToPerson = getRandomFromTable(scoreData.getConnectedToPerson());
+  score.value.twistOrComplication = getRandomFromTable(scoreData.getTwistOrComplication());
+
+  const clientType : Variant = generateRandomClientOrTargetType();
+  const targetType : Variant = generateRandomClientOrTargetType();
+  const workType : Variant = generateRandomWorkType();
+
+  setClientType(clientType);
+  setTargetType(targetType);
+  setWorkType(workType);
+}
+const setWorkType = (workType: Variant) => {
```

Comment:
Can work type be `rare` or `common`? If not it should be a different type. Not `Variant` i.e. `WorkTypeVariant`